internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -226,6 +237,17 @@ func setUnsupportedFormatCondition(image *metal3_v1alpha1.PreprovisioningImage)
 		reason, message)
 }
 
+func setMismatchedArchCondition(image *metal3_v1alpha1.PreprovisioningImage, infraArch string) {
+	message := fmt.Sprintf("PreprovisioningImage CPU architecture (%s) does not match InfraEnv CPU architecture (%s)", image.Spec.Architecture, infraArch)
+	reason := imageConditionReason(strcase.ToCamel(message))
```

Comment:
This turned out to be not such a great string.

I don't know what the conventions around this are though.
@eranco74 @avishayt @mhrivnak do any of you know if this `reason` should always be a camel cased string? Is there any reason for it to be created from the message?

FWIW this is what I'm getting when I test this live which feels less than ideal.
```
    - lastTransitionTime: "2022-10-26T18:01:37Z"
      message: PreprovisioningImage CPU architecture (x86_64) does not match InfraEnv
        CPU architecture (arm64)
      observedGeneration: 1
      reason: PreprovisioningImageCPUArchitecturex8664DoesNotMatchInfraEnvCPUArchitecturearm64
      status: "False"
      type: Ready
    - lastTransitionTime: "2022-10-26T18:01:37Z"
      message: PreprovisioningImage CPU architecture (x86_64) does not match InfraEnv
        CPU architecture (arm64)
      observedGeneration: 1
      reason: PreprovisioningImageCPUArchitecturex8664DoesNotMatchInfraEnvCPUArchitecturearm64
      status: "True"
      type: Error
```

Reply:
As mentioned on slack, the reason should be more concise.  Either InfraEnvArchitectureMismatch or InfraEnvArchMismatch is fine.